rasa_core/evaluate.py

```diff
@@ -324,13 +324,21 @@ def _predict_tracker_actions(tracker, agent, fail_on_prediction_errors=False,
                                                        events[:1],
                                                        agent.domain.slots)
 
+    in_training_data = True
+    test_in_training_data = True
+
     for event in events[1:]:
         if isinstance(event, ActionExecuted):
-            action_executed_result = \
+            action_executed_result, policy = \
                 _collect_action_executed_predictions(
                         processor, partial_tracker, event,
                         fail_on_prediction_errors
                 )
+            if (test_in_training_data and
```

Comment:
I think we should put this whole logic into a separate function and not into `_predict_tracker_actions `. I think it is fine to iterate twice over the events